controller/apps.go

```diff
@@ -205,7 +218,8 @@ func (c *AppsController) SetDeployment(ctx *app.SetDeploymentAppsContext) error
 		return witerrors.NewBadParameterError("podCount", "missing")
 	}
 
-	kc, err := c.getKubeClient(ctx)
+	kc, err := c.GetKubeClient(ctx)
+	defer cleanup(kc)
```

Comment:
Why call defer before error checking @aslakknutsen @ebaron ?

Reply:
In this case I would say it points to how the Client is implemented. It could get error after being opened. Defer just ensure to close it again.